lib/complete-provider.coffee

```diff
@@ -53,11 +58,19 @@ class CompleteProvider extends Provider
     'then', 'where'
   ]
 
-  initialize: (@editorView, @manager) ->
+  constructor: (@editor, @manager) ->
+    if (@editor.haskellCompletionProvider && @editor.haskellCompletionProvider != this)
+      # shouldn't happen, but log just in case
+      console.log "weird: editor already has a haskell completion provider"
+      @editor.haskellCompletionProvider.dispose()
```

Comment:
Having multiple completion providers allows the package to watch all open files.  A straightforward conversion of the code to a single provider just watches the currently active editor file; however, this introduces a bug; if you open a file (say "Module.hs"), make a change that should reflect in completion (like rename an exported module function), then open another file ("Main.hs") and use Atom's "Save All" command; the first file will save, but its completion database won't be rebuilt, because the provider isn't watching that file now.  The second file ("Main.hs") will then see the old name until you manually go back into the first file and save it while it is open in the editor, which forces the completion database to rebuild.  Suggestions on how to handle this?